### PR TITLE
Patch azidentity to upgrade MSAL requirement

### LINKED ISSUE #19871: New MSAL v0.8.0 breaks AZ Identity v1.2.0 in sdk for go
### Bug Report

This is similar to #19823 but appears to be caused by a new release of MSAL.  Here is what I'm seeing, and my analysis / workaround.

Last night MSAL for go v0.8.0 was released.  Since that time, when I build my application, I get the following errors (see below).

If I force use of MSAL v0.7.0 by adding `go mod edit -require github.com/AzureAD/microsoft-authentication-library-for-go@v0.7.0` to my makefile, the problem goes away.  And my project builds as it always has.

So the work-around works.  But it appears that there is an incompatibility between `github.com/AzureAD/microsoft-authentication-library-for-go@v0.8.0` and `github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0`

I'm using go v1.19.5 -- but I also tried with v1.18 with the same results.

Errors when built with MSAL v0.8.0:

```
# github.com/Azure/azure-sdk-for-go/sdk/azidentity
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/client_assertion_credential.go:52:44: cannot use c (variable of type confidential.Client) as type confidentialClient in struct literal:
        confidential.Client does not implement confidentialClient (wrong type for AcquireTokenByAuthCode method)
                have AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, opts ...confidential.AcquireByAuthCodeOption) (base.AuthResult, error)
                want AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...confidential.AcquireTokenByAuthCodeOption) (base.AuthResult, error)
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/client_certificate_credential.go:59:46: cannot use c (variable of type confidential.Client) as type confidentialClient in struct literal:
        confidential.Client does not implement confidentialClient (wrong type for AcquireTokenByAuthCode method)
                have AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, opts ...confidential.AcquireByAuthCodeOption) (base.AuthResult, error)
                want AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...confidential.AcquireTokenByAuthCodeOption) (base.AuthResult, error)
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/client_secret_credential.go:43:41: cannot use c (variable of type confidential.Client) as type confidentialClient in struct literal:
        confidential.Client does not implement confidentialClient (wrong type for AcquireTokenByAuthCode method)
                have AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, opts ...confidential.AcquireByAuthCodeOption) (base.AuthResult, error)
                want AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...confidential.AcquireTokenByAuthCodeOption) (base.AuthResult, error)
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/device_code_credential.go:85:66: cannot use c (variable of type public.Client) as type publicClient in struct literal:
        public.Client does not implement publicClient (wrong type for AcquireTokenByAuthCode method)
                have AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, opts ...public.AcquireByAuthCodeOption) (base.AuthResult, error)
                want AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...public.AcquireTokenByAuthCodeOption) (base.AuthResult, error)
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/device_code_credential.go:94:59: cannot use public.WithSilentAccount(c.account) (value of type interface{public.AcquireSilentOption; options.CallOption}) as type public.AcquireTokenSilentOption in argument to c.client.AcquireTokenSilent
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/interactive_browser_credential.go:62:60: cannot use c (variable of type public.Client) as type publicClient in struct literal:
        public.Client does not implement publicClient (wrong type for AcquireTokenByAuthCode method)
                have AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, opts ...public.AcquireByAuthCodeOption) (base.AuthResult, error)
                want AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...public.AcquireTokenByAuthCodeOption) (base.AuthResult, error)
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/interactive_browser_credential.go:70:59: cannot use public.WithSilentAccount(c.account) (value of type interface{public.AcquireSilentOption; options.CallOption}) as type public.AcquireTokenSilentOption in argument to c.client.AcquireTokenSilent
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/interactive_browser_credential.go:78:17: cannot use public.WithRedirectURI(c.options.RedirectURL) (value of type interface{public.AcquireInteractiveOption; options.CallOption}) as type public.InteractiveAuthOption in argument to append
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/managed_identity_credential.go:101:44: cannot use c (variable of type confidential.Client) as type confidentialClient in struct literal:
        confidential.Client does not implement confidentialClient (wrong type for AcquireTokenByAuthCode method)
                have AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, opts ...confidential.AcquireByAuthCodeOption) (base.AuthResult, error)
                want AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...confidential.AcquireTokenByAuthCodeOption) (base.AuthResult, error)
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/username_password_credential.go:46:85: cannot use c (variable of type public.Client) as type publicClient in struct literal:
        public.Client does not implement publicClient (wrong type for AcquireTokenByAuthCode method)
                have AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, opts ...public.AcquireByAuthCodeOption) (base.AuthResult, error)
                want AcquireTokenByAuthCode(ctx context.Context, code string, redirectURI string, scopes []string, options ...public.AcquireTokenByAuthCodeOption) (base.AuthResult, error)
/go/pkg/mod/github.com/!azure/azure-sdk-for-go/sdk/azidentity@v1.2.0/username_password_credential.go:46:85: too many errors
```

### PR DESCRIPTION
**Pull request by chlowell**
MSAL v0.8.0 breaks the test fakes we wrote against prior versions. This patch simply upgrades to the latest MSAL and fixes these fakes so azidentity users are able to upgrade MSAL.

I cherry-picked the commit that removed `.../resourcemanager/loadtestservice/armloadtestservice` because that module existed in the repo at this point in history but has since been retracted, causing our dependency check to fail.

### CODE REVIEW

**Comment by benbp** [comment]
@chlowell should I go ahead and delete the loadtestservice pipelines?

**Comment by jhendrixMSFT** [comment]
@benbp I would presume we want branch protection rules in place for `hotfix/**` branches given that we'll be releasing bits from them.

Also, I wonder if the branch should be named `hotfix/azidentity-1.2` as it would allow multiple patch releases (the current naming implies there would be multiple patch branches).

File: sdk/azidentity/interactive_browser_credential_test.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func TestInteractiveBrowserCredential_RedirectURLLive(t *testing.T) {
 		t.Skip("set AZIDENTITY_RUN_MANUAL_BROWSER_TESTS to run this test")
 	}
 	// the default application's registration allows redirecting to any localhost port
-	cred, err := NewInteractiveBrowserCredential(&InteractiveBrowserCredentialOptions{RedirectURL: "localhost:8180"})
+	cred, err := NewInteractiveBrowserCredential(&InteractiveBrowserCredentialOptions{RedirectURL: "http://localhost:8180"})
```
Comment (jhendrixMSFT) [inline sdk/azidentity/interactive_browser_credential_test.go:82]: Was there a change in behavior that requires the scheme to be specified now?

**Comment by benbp** [comment]
@jhendrixMSFT updated to mirror the `release/**/*` rules

**Comment by chlowell** [comment]
> Also, I wonder if the branch should be named `hotfix/azidentity-1.2` as it would allow multiple patch releases (the current naming implies there would be multiple patch branches).

Do we need to keep the branch? I planned to delete it after the release. If we need to make v1.2.2, we can branch off the tag for v1.2.1.

**Comment by jhendrixMSFT** [comment]
That's a good question.  If the commit is tagged and the branch is deleted, is the tag enough to keep the commit from evaporating?

**Comment by jhendrixMSFT** [comment]
Also, won't this branch need to be merged back into `main`, or is the fix already in `main`?

**Comment by chlowell** [comment]
> That's a good question. If the commit is tagged and the branch is deleted, is the tag enough to keep the commit from evaporating?

Yes, though it's perhaps a moot point because this PR's target branch is protected. @benbp what's the rationale for protecting hotfix/release branches?

> Also, won't this branch need to be merged back into `main`, or is the fix already in `main`?

`main` already has the upgrade.

**Comment by jhendrixMSFT** [comment]
Branch protection was at my request as I thought the branch would have to live in perpetuity.  If that's not the case then it's unnecessary (sorry for the confusion).

**Comment by benbp** [comment]
I assumed branch protection was to make sure reviews happened for hotfix commits.

I just verified, if you delete a branch, any tags/tagged commits remain.

**Comment by jhendrixMSFT** [comment]
Good point about ensuring reviews happen before committing.

**Pull request merged**